src/main/scala/li/cil/oc/server/machine/luac/LuaStateFactory.scala

```diff
@@ -137,7 +137,12 @@ abstract class LuaStateFactory {
   // loaded at the time the initializer runs.
   private def prepareLoad(lib: String): Unit = jnlua.NativeSupport.getInstance().setLoader(new Loader {
     def load(): Unit = {
-      System.load(lib)
+      // This redirects to `System.load(lib)`. It is required because
+      // Java expects the correct ClassLoader to also load the native
+      // library. Simply calling load here doesn't work in dev,
+      // because the JNLua jar only gets merged when the mod is built.
+      // Production environments are unaffected.
+      jnlua.NativeSupport.proxyLoad(lib)
```

Comment:
Any changes to JNLua will need to go into a separate PR: https://github.com/MightyPirates/OC-JNLua - we actually have a consistent build system for it now.

Reply:
I've integrated the new lua build pipeline, and made the development environment run from a mod jar (like FancyGradle does in the 1.12 branch) and so a PR is no longer necessary.

Reply:
Excellent!